xUnit Migration: Assert.Same converted to non-existent IsSameReference method

## Description

When using the TUnit xUnit migration code fixer (TUXU0001), the fixer converts xUnit's `Assert.Same()` to `.IsSameReference()` and `Assert.NotSame()` to `.IsNotSameReference()`, but these methods don't appear to exist in TUnit, resulting in CS1061 compilation errors.

## Steps to Reproduce

1. Create an xUnit test using `Assert.Same()`:
```csharp
using Xunit;

[Fact]
public void Test_Reference_Equality()
{
    var expected = new object();
    var actual = expected;
    
    Assert.Same(expected, actual);
    Assert.NotSame(expected, new object());
}
```

2. Install TUnit package and run `dotnet format analyzers --severity info --diagnostics TUXU0001`

## Expected Behavior

The code should be converted to valid TUnit syntax, such as:
```csharp
[Test]
public async Task Test_Reference_Equality()
{
    var expected = new object();
    var actual = expected;
    
    await Assert.That(actual).IsSameAs(expected);  // Or whatever the correct TUnit method is
    await Assert.That(new object()).IsNotSameAs(expected);
}
```

## Actual Behavior

The code is converted to:
```csharp
[Test]
public async Task Test_Reference_Equality()
{
    var expected = new object();
    var actual = expected;
    
    await Assert.That(actual).IsSameReference(expected);
    await Assert.That(new object()).IsNotSameReference(expected);
}
```

This causes:
```
error CS1061: 'ValueAssertion<object>' does not contain a definition for 'IsSameReference' and no accessible extension method 'IsSameReference' accepting a first argument of type 'ValueAssertion<object>' could be found
```

## Example from Moq Test Suite

Original:
```csharp
[Fact]
public void Callback_accepts_instance_method_as_callback()
{
    var instance = new Instance();
    Action<int> callback = instance.Action;
    Assert.Single(callback.Method.GetParameters());
    Assert.Same(instance, callback.Target);

    this.setup.Callback(callback);
}
```

Converted (incorrectly):
```csharp
[Test]
public async Task Callback_accepts_instance_method_as_callback()
{
    var instance = new Instance();
    Action<int> callback = instance.Action;
    await Assert.That(callback.Method.GetParameters()).HasSingleItem();
    await Assert.That(callback.Target).IsSameReference(instance);

    this.setup.Callback(callback);
}
```

## Impact

This affects every test that uses `Assert.Same()` or `Assert.NotSame()`. In the Moq test suite, this resulted in 70+ compilation errors across various types.

## Environment

- TUnit Version: 1.9.91
- .NET SDK: 10.0.101
- OS: Windows

--- a/TUnit.Analyzers.CodeFixers/XUnitMigrationCodeFixProvider.cs
+++ b/TUnit.Analyzers.CodeFixers/XUnitMigrationCodeFixProvider.cs
@@ -569,9 +569,9 @@ public class XUnitMigrationCodeFixProvider : BaseMigrationCodeFixProvider
 
                 // Reference assertions
                 "Same" when arguments.Count >= 2 =>
-                    CreateTUnitAssertion("IsSameReference", arguments[1].Expression, arguments[0]),
+                    CreateTUnitAssertion("IsSameReferenceAs", arguments[1].Expression, arguments[0]),
                 "NotSame" when arguments.Count >= 2 =>
-                    CreateTUnitAssertion("IsNotSameReference", arguments[1].Expression, arguments[0]),
+                    CreateTUnitAssertion("IsNotSameReferenceAs", arguments[1].Expression, arguments[0]),
 
                 // String/Collection contains
                 "Contains" when arguments.Count >= 2 =>

--- a/TUnit.Analyzers.Tests/XUnitMigrationAnalyzerTests.cs
+++ b/TUnit.Analyzers.Tests/XUnitMigrationAnalyzerTests.cs
@@ -861,6 +861,82 @@ public class XUnitMigrationAnalyzerTests
             );
     }
 
+    [Test]
+    public async Task Assert_Same_Can_Be_Converted()
+    {
+        await CodeFixer
+            .VerifyCodeFixAsync(
+                """
+                {|#0:using TUnit.Core;
+
+                public class MyClass
+                {
+                    [Fact]
+                    public void MyTest()
+                    {
+                        var expected = new object();
+                        var actual = expected;
+                        Assert.Same(expected, actual);
+                    }
+                }|}
+                """,
+                Verifier.Diagnostic(Rules.XunitMigration).WithLocation(0),
+                """
+                using TUnit.Core;
+
+                public class MyClass
+                {
+                    [Test]
+                    public async Task MyTest()
+                    {
+                        var expected = new object();
+                        var actual = expected;
+                        await Assert.That(actual).IsSameReferenceAs(expected);
+                    }
+                }
+                """,
+                ConfigureXUnitTest
+            );
+    }
+
+    [Test]
+    public async Task Assert_NotSame_Can_Be_Converted()
+    {
+        await CodeFixer
+            .VerifyCodeFixAsync(
+                """
+                {|#0:using TUnit.Core;
+
+                public class MyClass
+                {
+                    [Fact]
+                    public void MyTest()
+                    {
+                        var obj1 = new object();
+                        var obj2 = new object();
+                        Assert.NotSame(obj1, obj2);
+                    }
+                }|}
+                """,
+                Verifier.Diagnostic(Rules.XunitMigration).WithLocation(0),
+                """
+                using TUnit.Core;
+
+                public class MyClass
+                {
+                    [Test]
+                    public async Task MyTest()
+                    {
+                        var obj1 = new object();
+                        var obj2 = new object();
+                        await Assert.That(obj2).IsNotSameReferenceAs(obj1);
+                    }
+                }
+                """,
+                ConfigureXUnitTest
+            );
+    }
+
     private static void ConfigureXUnitTest(Verifier.Test test)
     {
         var globalUsings = ("GlobalUsings.cs", SourceText.From("global using Xunit;"));

